crates/sui-types/src/crypto.rs

```diff
@@ -1461,7 +1497,8 @@ impl SignatureScheme {
         match self {
             SignatureScheme::ED25519 => 0x00,
             SignatureScheme::Secp256k1 => 0x01,
-            SignatureScheme::BLS12381 => 0xff,
+            SignatureScheme::Secp256r1 => 0x02,
+            SignatureScheme::BLS12381 => 0x03,
```

Comment:
ditto

Reply:
I discovered an incorrect test setup that the BLS signature scheme is actually used (with the 0xff value set). this is used to compute. 

```
impl From<&AuthorityPublicKeyBytes> for SuiAddress {
    fn from(pkb: &AuthorityPublicKeyBytes) -> Self {
        let mut hasher = Sha3_256::default();
        hasher.update([AuthorityPublicKey::SIGNATURE_SCHEME.flag()]);
        hasher.update(pkb);
        let g_arr = hasher.finalize();

        let mut res = [0u8; SUI_ADDRESS_LENGTH];
        // OK to access slice because Sha3_256 should never be shorter than SUI_ADDRESS_LENGTH.
        res.copy_from_slice(&AsRef::<[u8]>::as_ref(&g_arr)[..SUI_ADDRESS_LENGTH]);
        SuiAddress(res)
    }
}
```

An authority address is not derived from the protocol key (BLS) but from the account key (Ed25519). This involves a larger fix out of scope so I will follow up a PR

Reply:
@joyqvq awesome finding (let's create a bug issue), I guess we still need to remove `SignatureScheme::BLS12381 => 0x03` from this PR?

Reply:
yeah, ill keep it as 0xff as it is now.